src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -92,15 +92,31 @@ def register_submission(
             objecttype_url = objecttype["url"]
 
         with get_objects_client(options["objects_api_group"]) as objects_client:
-            response = execute_unless_result_exists(
-                partial(
-                    objects_client.create_object,
-                    objecttype_url=objecttype_url,
-                    record_data=record_data,
-                ),
-                submission,
-                "intermediate.objects_api_object",
-            )
+            # update or create the object
+            if not (
+                options.get("update_existing_object")
+                and submission.initial_data_reference
+            ):
```

Comment:
Is this condition right? I would expect the opposite:

```suggestion
            if (
                options.get("update_existing_object")
                and submission.initial_data_reference
            ):
                # update object
            else:
                # create object
```

Reply:
I don't see the difference here. I understand that we are doing the same check in your suggestion but we start from the update. Do you mean it's better in this order?